### PR TITLE
feat: channel list redesign

### PR DESCRIPTION
**Pull request by khushal87**
This pull request introduces significant improvements to the channel preview and messaging experience, focusing on richer message previews, enhanced delivery status indicators, and improved flexibility in UI customization. The changes modernize the `ChannelPreview` and `ChannelList` components to better support draft messages, typing indicators, poll previews, and message delivery statuses, while also providing more control over the layout and appearance of muted status indicators.

**Enhancements to Channel Preview and Message Status:**

* The `ChannelPreview` and related components now support richer message previews, including draft messages, typing indicators, poll previews, and error states. Message delivery status (sent, delivered, read) is now visually indicated for the latest message sent by the current user. [[1]](diffhunk://#diff-3baf806204bf22f2aca161a9ca9bb0d1593e82e7bdbd22dbc30e99ebf957b18aL1-R219) [[2]](diffhunk://#diff-525ac0e781471046d4c53d7f607e02ddab3f96b2aff39f23fd001f80cd1f7bddR1-R108) [[3]](diffhunk://#diff-3581be452d8577eafae622cc00bf5e14178b0d2faa4b68cd678fd72153fa5f5fL30-R32) [[4]](diffhunk://#diff-acc1867b62b2fdd1e5a06bef695f25e718ffd41f649e3b519d14a201ba05cf50L1-L44) [[5]](diffhunk://#diff-acc1867b62b2fdd1e5a06bef695f25e718ffd41f649e3b519d14a201ba05cf50R32-L82) [[6]](diffhunk://#diff-acc1867b62b2fdd1e5a06bef695f25e718ffd41f649e3b519d14a201ba05cf50R48-L103) [[7]](diffhunk://#diff-acc1867b62b2fdd1e5a06bef695f25e718ffd41f649e3b519d14a201ba05cf50R65-R138)

**UI Customization and Layout Improvements:**

* The `mutedStatusPosition` prop is added throughout the channel list and preview components, allowing the muted status indicator to be positioned either inline with the title or at the bottom of the preview. This prop is now part of the `ChannelListProps` and is passed through context to all relevant components. [[1]](diffhunk://#diff-03113275ce71ed42ffadaa51dcfa09e590276e28cae0382e11ed7d43211ec2fdR58) [[2]](diffhunk://#diff-03113275ce71ed42ffadaa51dcfa09e590276e28cae0382e11ed7d43211ec2fdR289) [[3]](diffhunk://#diff-03113275ce71ed42ffadaa51dcfa09e590276e28cae0382e11ed7d43211ec2fdR421) [[4]](diffhunk://#diff-d53cca04352d1491080f14566ad673785d265be0720c68cced77e573be56f7caR38) [[5]](diffhunk://#diff-d53cca04352d1491080f14566ad673785d265be0720c68cced77e573be56f7caR84) [[6]](diffhunk://#diff-d53cca04352d1491080f14566ad673785d265be0720c68cced77e573be56f7caR96) [[7]](diffhunk://#diff-acc1867b62b2fdd1e5a06bef695f25e718ffd41f649e3b519d14a201ba05cf50R32-L82) [[8]](diffhunk://#diff-acc1867b62b2fdd1e5a06bef695f25e718ffd41f649e3b519d14a201ba05cf50R65-R138)

**Visual and Interaction Tweaks:**

* The channel preview UI is updated to use `Pressable` instead of `TouchableOpacity` for better feedback and flexibility, and the avatar size is increased for improved visibility. Layout and styling are modernized for consistency and accessibility. [[1]](diffhunk://#diff-acc1867b62b2fdd1e5a06bef695f25e718ffd41f649e3b519d14a201ba05cf50L1-L44) [[2]](diffhunk://#diff-acc1867b62b2fdd1e5a06bef695f25e718ffd41f649e3b519d14a201ba05cf50R65-R138)

**Minor UI Adjustments:**

* The badge size in the image overlay component is reduced from large to small for a more compact appearance.
* The `UserAdd` icon in the new direct messaging screen is now explicitly sized for consistency.

### CODE REVIEW

File: package/src/components/ChannelList/Skeleton.tsx [hunk 1]
```diff
@@ -1,52 +1,24 @@
...
-import { Defs, LinearGradient, Path, Rect, Stop, Svg } from 'react-native-svg';
+import Svg, { Path, Rect, Defs, LinearGradient, Stop, ClipPath, G, Mask } from 'react-native-svg';
 
 import { useTheme } from '../../contexts/themeContext/ThemeContext';
 
-const paddingLarge = 16;
-const paddingMedium = 12;
-const paddingSmall = 8;
-
-const AnimatedPath = Animated.createAnimatedComponent(Path);
```
Comment (isekovanic) [inline package/src/components/ChannelList/Skeleton.tsx:20]: Are we no longer animating this ? What's the reason behind it ?

File: package/src/components/ChannelPreview/hooks/__tests__/useLatestMessagePreview.test.tsx [hunk 2]
```diff
@@ -1,222 +0,0 @@
...
-const initChannelFromData = async (chatClient: StreamChat, channelData: ChannelResponse) => {
-  const mockedChannel = generateChannelResponse(channelData);
-  useMockedApis(chatClient, [getOrCreateChannelApi(mockedChannel)]);
-  const channel = chatClient.channel('messaging', mockedChannel.channel.id);
-  await channel.watch();
-
-  return channel;
-};
-
-describe('useLatestMessagePreview', () => {
```
Comment (isekovanic) [inline package/src/components/ChannelPreview/hooks/__tests__/useLatestMessagePreview.test.tsx:35]: Why are we deleting a bunch of tests ? I can't see them moved elsewhere (might be wrong though)

Comment (khushal87) [reply package/src/components/ChannelPreview/hooks/__tests__/useLatestMessagePreview.test.tsx:35]: This hook is not relevant any more so its deleted. We have to properly write tests for `MessagePreview` component. I have it in my todo and there are a lot of edge cases, so I will handle it later. If you want I can spend time to add now. But the reason this is deleted, is the hook is non-relevant anymore.

File: package/src/components/ChannelList/Skeleton.tsx (hunk 1 above)
Comment (khushal87) [reply package/src/components/ChannelList/Skeleton.tsx:20]: Its animated but this is not required. I have added a simple SVG and the Animated view applies to the view. You can run the app to see it working.

File: package/src/components/ThreadList/ThreadListItem.tsx [hunk 3]
```diff
@@ -85,169 +70,6 @@ export const attachmentTypeIconMap = {
   voiceRecording: '🎙️',
 } as const;
 
-type LatestMessage = ReturnType<ChannelState['formatMessage']> | MessageResponse;
```
Comment (isekovanic) [inline package/src/components/ThreadList/ThreadListItem.tsx:88]: Why are we doing changes to `ThreadList` in a PR dedicated to `ChannelList` ?

Comment (khushal87) [reply package/src/components/ThreadList/ThreadListItem.tsx:88]: To fix the build errors for now as MessagePreview has a different prop signature now. I will take care of it in ThreadList soon. Don't worry, I will make sure nothing would be broken.

File: package/src/i18n/fr.json [hunk 4]
```diff
@@ -148,6 +148,14 @@
   "{{ index }} of {{ photoLength }}": "{{ index }} sur {{ photoLength }}",
   "{{ replyCount }} Replies": "{{ replyCount }} Réponses",
   "{{ user }} is typing": "{{ user }} est en train d'écrire",
+  "You voted: {{ option }}": "Vous avez voté: {{ option }}",
```
Comment (isekovanic) [inline package/src/i18n/fr.json:151]: Are we also removing the old translations which are no longer relevant ?

File: package/src/icons/Mute.tsx [hunk 5]
```diff
@@ -1,23 +1,20 @@
...
-  <Svg viewBox='0 0 20 20' fill='none' {...props}>
-    <Mask id='path-1-inside-1_5641_10062' fill='white'>
-      <Path
-        fillRule='evenodd'
-        clipRule='evenodd'
-        d='M14.1667 4.94994V3.52203C14.1667 2.67296 13.2054 2.18057 12.5163 2.67668L8.40442 5.63728C8.22712 5.76492 8.01421 5.8336 7.79576 5.8336H6.45833C5.19268 5.8336 4.16666 6.85961 4.16666 8.12526V11.8752C4.16666 12.7088 4.61171 13.4384 5.27714 13.8395L2.68306 16.4336C2.43898 16.6777 2.43898 17.0733 2.68306 17.3174C2.92713 17.5615 3.32286 17.5615 3.56694 17.3174L17.3169 3.56743C17.561 3.32335 17.561 2.92762 17.3169 2.68355C17.0728 2.43947 16.6772 2.43947 16.4331 2.68355L14.1667 4.94994ZM6.22574 12.8909C5.76241 12.7852 5.41666 12.3706 5.41666 11.8752V8.12526C5.41666 7.54997 5.88303 7.0836 6.45833 7.0836H7.79576C8.27634 7.0836 8.74475 6.9325 9.13475 6.6517L12.9167 3.92875V6.19994L6.22574 12.8909Z'
-      />
-      <Path d='M12.9167 16.0718L9.23183 13.4187L8.33391 14.3167C8.35791 14.3312 8.3815 14.3467 8.40441 14.3632L12.5163 17.3238C13.2054 17.82 14.1667 17.3276 14.1667 16.4785V8.4839L12.9167 9.7339V16.0718Z' />
-    </Mask>
+export const Mute = ({ height, width, ...rest }: IconProps) => (
```
Comment (isekovanic) [inline package/src/icons/Mute.tsx:7]: Changing this also affects other parts of the code, such as the quick actions menu in the attachment picker. Has this been checked ?

Comment (khushal87) [reply package/src/icons/Mute.tsx:7]: Yeah I fixed in in the actions menu atleast.

File: package/src/i18n/fr.json (hunk 4 above)
Comment (khushal87) [reply package/src/i18n/fr.json:151]: Not yet, we can do that cleanup later. Once we know what's not used finally.

Comment (khushal87) [reply package/src/i18n/fr.json:151]: I can run over a check later to remove onces that are not used.

File: package/src/components/ChannelPreview/ChannelPreviewMessage.tsx [hunk 6]
```diff
@@ -1,38 +1,206 @@
...
+  const { text: draftText } = useStateStore(
+    channel.messageComposer.textComposer.state,
+    textComposerStateSelector,
+  );
 
-  return (
-    <View style={[styles.container, container]}>
-      <MessagePreview previews={latestMessagePreview.previews} />
-    </View>
+  const { attachments } = useStateStore(
```
Comment (isekovanic) [inline package/src/components/ChannelPreview/ChannelPreviewMessage.tsx]: Please extract this in a separate hook. There's no reason for `ChannelPreviewMessage` to be updating every time the attachments change in `messageComposer`. These need to be decoupled.

File: package/src/components/ChannelPreview/ChannelPreviewMessage.tsx [hunk 7]
```diff
@@ -1,38 +1,206 @@
...
+  const membersWithoutSelf = useMemo(() => {
+    return Object.values(channel.state.members).filter(
+      (member) => member.user?.id !== client.user?.id,
+    );
+  }, [channel.state.members, client.user?.id]);
+
+  const isFailedMessage =
+    lastMessage?.status === MessageStatusTypes.FAILED || lastMessage?.type === 'error';
+
+  const renderMessagePreview = (message: LocalMessage | MessageResponse | DraftMessage) => {
```
Comment (isekovanic) [inline package/src/components/ChannelPreview/ChannelPreviewMessage.tsx]: The identity of this render function changes basically on every render, causing a bad cascading effect. Let's please fix this.

File: package/src/components/ChannelPreview/ChannelPreviewMessage.tsx [hunk 8]
```diff
@@ -1,38 +1,206 @@
...
+    return (
+      <MessagePreview
+        message={message}
+        textStyle={[
+          styles.subtitle,
+          { color: message?.type === 'deleted' ? semantics.textTertiary : semantics.textSecondary },
+        ]}
+        iconProps={{
```
Comment (isekovanic) [inline package/src/components/ChannelPreview/ChannelPreviewMessage.tsx]: Passing `iconProps` inline like so invalidates any potential memoization you've included downstream. In other words, every single `React.memo` that's added there is for nothing

File: package/src/components/MessagePreview/MessagePreview.tsx [hunk 9]
```diff
@@ -1,67 +1,355 @@
...
+    style?: StyleProp<TextStyle>;
+  }) => {
+    const { client } = useChatContext();
+    const poll = client.polls.fromState(message?.poll_id ?? '');
+    const { name: pollName } = useStateStore(poll?.state, selector) ?? {};
+    const styles = useStyles();
+
+    const subtitle = useMemo(() => {
+      const attachments = message?.attachments;
+      const giphyAttachments = attachments?.filter(
```
Comment (isekovanic) [inline package/src/components/MessagePreview/MessagePreview.tsx]: Even though `attachments` will always have a very limited number of stuff inside, I'm really against this approach. You're allocating 6 arrays, just to check if `arr.length > 0` further down. This is all work for the garbage collector going forward, for every `ChannelPreview`, all the time. Can we at the very least do something a bit more optimal and `.reduce` a `Map` or an `Set` which tells us what types of attachments this `message` has ? We can even add it to some general purpose hook or whatever.

Comment (isekovanic) [reply package/src/components/MessagePreview/MessagePreview.tsx]: (and it will automatically work for pretty much anything going forward)

Comment (isekovanic) [reply package/src/components/MessagePreview/MessagePreview.tsx]: PS: I just saw that this is how it was done before as well, so just to be clear - not saying it was introduced in this PR but let's maybe simplify it

Comment (khushal87) [reply package/src/components/MessagePreview/MessagePreview.tsx]: Done

File: package/src/components/ChannelPreview/ChannelPreviewMessage.tsx (hunk 8 above)
Comment (khushal87) [reply package/src/components/ChannelPreview/ChannelPreviewMessage.tsx]: Done

File: package/src/components/ChannelPreview/ChannelPreviewMessage.tsx (hunk 7 above)
Comment (khushal87) [reply package/src/components/ChannelPreview/ChannelPreviewMessage.tsx]: Done

File: package/src/components/ChannelPreview/ChannelPreviewMessage.tsx (hunk 6 above)
Comment (khushal87) [reply package/src/components/ChannelPreview/ChannelPreviewMessage.tsx]: Done

File: package/src/components/ChannelPreview/hooks/__tests__/useLatestMessagePreview.test.tsx (hunk 2 above)
Comment (isekovanic) [reply package/src/components/ChannelPreview/hooks/__tests__/useLatestMessagePreview.test.tsx:35]: Hmm okay, and is there a hook that replaces it ? Or what's the deal ?

File: package/src/components/ThreadList/ThreadListItem.tsx (hunk 3 above)
Comment (isekovanic) [reply package/src/components/ThreadList/ThreadListItem.tsx:88]: Yes but does this break anything right now ? If not, then you can resolve this comment.

File: package/src/components/ChannelPreview/hooks/__tests__/useLatestMessagePreview.test.tsx (hunk 2 above)
Comment (khushal87) [reply package/src/components/ChannelPreview/hooks/__tests__/useLatestMessagePreview.test.tsx:35]: MessagePreview replaces it

File: package/src/components/ThreadList/ThreadListItem.tsx (hunk 3 above)
Comment (khushal87) [reply package/src/components/ThreadList/ThreadListItem.tsx:88]: Nope

File: package/src/i18n/fr.json (hunk 4 above)
Comment (isekovanic) [reply package/src/i18n/fr.json:151]: Okay, but we're leaving a lot of stuff to do "later". It worries me a bit that we'll have too much on our plate to clean up later on and it'll introduce issues. Right now we're just bloating the SDK with redundant stuff.

File: package/src/components/ChannelPreview/ChannelPreviewMessage.tsx [hunk 10]
```diff
@@ -1,38 +1,176 @@
...
+  }, [channel.state.members, client.user?.id]);
+
+  const isFailedMessage =
+    lastMessage?.status === MessageStatusTypes.FAILED || lastMessage?.type === 'error';
+
+  const color = useMemo(() => {
+    return lastMessage?.type === 'deleted' ? semantics.textTertiary : semantics.textSecondary;
+  }, [semantics.textTertiary, semantics.textSecondary, lastMessage?.type]);
+
+  const textStyle = useMemo(() => {
```
Comment (isekovanic) [inline package/src/components/ChannelPreview/ChannelPreviewMessage.tsx:56]: This should come from `useStyles` preferably, we're doing an extra `useMemo` for no reason here.

For `iconProps` it's a bit different, but we can think about fixing this later I guess.

File: package/src/components/ChannelPreview/hooks/__tests__/useLatestMessagePreview.test.tsx (hunk 2 above)
Comment (isekovanic) [reply package/src/components/ChannelPreview/hooks/__tests__/useLatestMessagePreview.test.tsx:35]: In that case instead of using a hook (which returns an atomic string) we're relying on the component itself deciding. I would definitely move this inside of a hook to make sure that updates do not happen where they should not

File: package/src/components/ChannelPreview/ChannelPreviewMessage.tsx (hunk 10 above)
Comment (khushal87) [reply package/src/components/ChannelPreview/ChannelPreviewMessage.tsx:56]: Done

**Review by isekovanic — APPROVED** [review]
